pkg/noderesourcetopology/plugin.go

```diff
@@ -96,19 +126,27 @@ func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error)
 		return nil, err
 	}
 
-	scoringFunction, err := getScoringStrategyFunction(tcfg.ScoringStrategy.Type)
-	if err != nil {
-		return nil, err
-	}
-
 	resToWeightMap := make(resourceToWeightMap)
 	for _, resource := range tcfg.ScoringStrategy.Resources {
 		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
 	}
 
+	var scoringHandlers scoreHandlersMap
+
+	if tcfg.ScoringStrategy.Type == apiconfig.LeastNUMANodes {
+		scoringHandlers = leastNUMAscoreHandlers()
+	} else {
+		strategy, err := getScoringStrategyFunction(tcfg.ScoringStrategy.Type)
```

Comment:
(continue the comment from https://github.com/kubernetes-sigs/scheduler-plugins/pull/474#discussion_r1083109506)

I _thought_  scheduler-plugin has the logic in place for plugin arguments, but seems not. Fine with current logic.

A general follow-up can be done to catch all illegal plugin arguments like https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/validation/validation_pluginargs.go.